src/core/services/theming/theming.js

```diff
@@ -589,10 +579,41 @@ function generateThemes($injector) {
       }
     });
   }
+}
+
+function generateTheme(name) {
+  var theme = THEMES[name];
+  var head = document.head;
+  var firstChild = head ? head.firstElementChild : null;
+
+  if (!GENERATED[name]) {
+    // For each theme, use the color palettes specified for
+    // `primary`, `warn` and `accent` to generate CSS rules.
+    THEME_COLOR_TYPES.forEach(function(colorType) {
+      var styleStrings = parseRules(theme, colorType, rulesByType[colorType]);
+      while (styleStrings.length) {
+        var styleContent = styleStrings.shift();
+        if (styleContent) {
```

Comment:
FYI: I added this `if (styleContent)` check because the code was adding a whole bunch of empty `<style>` elements for some reason. I didn't take the time to go figure out why.